modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -468,47 +472,124 @@ TensorMechanicsAction::actOutputGeneration()
             {
-              type = "RankTwoScalarAux";
+              type = "RankTwoTensorScalar";
               params = _factory.getValidParams(type);
-              params.set<MaterialPropertyName>("rank_two_tensor") = _base_name + r2a->second;
-              params.set<MooseEnum>("scalar_type") = r2sa.second.first;
+              params.set<MaterialPropertyName>("rank_two_tensor") = _base_name + r2q->second;
+              params.set<MooseEnum>("scalar_type") = r2sq.second.first;
+            }
+            else
+              mooseWarning("Internal error. The permitted tensor shortcuts in "
+                           "'_ranktwo_tensor_scalar_table' must be keys in the "
+                           "'_ranktwo_tensor_component_table'.");
+          }
+
+      //
+      // RankTwoScalarDirectionalComponent
+      for (const auto & r2sdq : _ranktwo_scalar_directional_component_table)
+        for (const auto & t : r2sdq.second.second)
+          if (r2sdq.first + '_' + t == out)
+          {
+            const auto r2q = _ranktwo_tensor_component_table.find(t);
+            if (r2q != _ranktwo_tensor_component_table.end())
+            {
+
+              type = "RankTwoScalarDirectionalComponent";
+              params = _factory.getValidParams(type);
+              params.set<MaterialPropertyName>("rank_two_tensor") = _base_name + r2q->second;
+              params.set<MooseEnum>("scalar_type") = r2sdq.second.first;
             }
             else
               mooseError("Internal error. The permitted tensor shortcuts in "
-                         "'_ranktwoscalaraux_table' must be keys in the '_ranktwoaux_table'.");
+                         "'_ranktwo_scalar_directional_component_table' must be keys in the "
+                         "'_ranktwo_tensor_component_table'.");
           }
 
-      if (type != "")
+      //
+      // RankTwoCylindricalComponent
+      for (const auto & r2sdq : _ranktwo_cylindrical_component_table)
+        for (const auto & t : r2sdq.second.second)
+          if (r2sdq.first + '_' + t == out)
+          {
+            const auto r2q = _ranktwo_tensor_component_table.find(t);
+            if (r2q != _ranktwo_tensor_component_table.end() &&
+                _coord_system != Moose::COORD_RSPHERICAL)
+            {
+
+              type = "RankTwoCylindricalComponent";
+              params = _factory.getValidParams(type);
+              params.set<MaterialPropertyName>("rank_two_tensor") = _base_name + r2q->second;
+              params.set<MooseEnum>("scalar_type") = r2sdq.second.first;
+            }
+            else
+              mooseError("Internal error. The permitted tensor shortcuts in "
+                         "'_ranktwo_cylindrical_component_table' must be keys in the "
+                         "'_ranktwo_tensor_component_table'.");
+          }
+      //
+      // This material property is already created by other materials such as
+      // ZryCreepLimbackHoppeUpdate.
```

Comment:
That's a BISON model name.  Don't reference here.  You could just say 'by creep or plasticity models'.